static/app/components/core/button/index.tsx

```diff
@@ -93,14 +93,10 @@ type ElementProps<E> = Omit<React.ButtonHTMLAttributes<E>, 'label' | 'size' | 't
 
 export interface BaseButtonProps
   extends CommonButtonProps,
-    ElementProps<HTMLButtonElement | HTMLAnchorElement> {
-  download?: HTMLAnchorElement['download'];
-  external?: boolean;
-  href?: string;
-  preventScrollReset?: boolean;
-  ref?: React.Ref<HTMLButtonElement | HTMLAnchorElement>;
-  replace?: boolean;
-  to?: string | LocationDescriptor;
+    ElementProps<HTMLButtonElement> {
+  href?: never;
```

Comment:
I will remove these in a next PR, as there's something going on where TS will still allow these to be passed in some instances if these are not typed as never.

Reply:
I think it’s good to have them as a union descriminator